File: AN_2/SEM_2/MPP/java/lab4/src/main/java/ro/mpp/labfx/service/CursaService.java
package ro.mpp.labfx.service;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ro.mpp.labfx.domain.Cursa;
import ro.mpp.labfx.domain.CursaDTO;
import ro.mpp.labfx.repository.CursaRepository;
import ro.mpp.labfx.repository.ParticipantCursaRepository;

import java.util.ArrayList;
import java.util.List;

public class CursaService {

    private CursaRepository cursaRepository;
    private ParticipantCursaRepository participantCursaRepository;

    public CursaService(CursaRepository cursaRepository, ParticipantCursaRepository participantCursaRepository) {
        this.cursaRepository = cursaRepository;
        this.participantCursaRepository = participantCursaRepository;
    }

    public List<Cursa> findAll() {
      return (List<Cursa>) cursaRepository.findAll();
    }

    public List<CursaDTO> listDTOcurse (List<Cursa> curse, ParticipantCursaService participantCursaService) {
        if (curse == null || curse.isEmpty()) {
            return List.of();
        }

        List<CursaDTO> lista = new ArrayList<>();
        for (Cursa c : curse) {
            int nr_part = participantCursaService.findNrPartByCursa(c);
            CursaDTO dto = new CursaDTO(c.getNume(),c.getCapacitateMinima(),c.getCapacitateMaxima(), nr_part);
            lista.add(dto);
        }
        return lista;
    }


}

File: AN_2/SEM_2/MPP/c#/lab4/lab4/service/CursaService.cs
using lab4.domain;
using log4net;

namespace lab4.service;

public class CursaService(ICursaRepository cursaRepo, ParticipantCursaService participantCursaService)
{
    private static readonly ILog Logger = LogManager.GetLogger(typeof(CursaService));
    private readonly ParticipantCursaService _participantCursaService = participantCursaService;

    public List<Cursa> FindAllCurse()
    {
        Logger.Debug("Incep cautarea tuturor curselor.");
        return cursaRepo.FindAll();
    }
    
    public List<CursaDto> ListDtoCurse(ParticipantCursaService participantCursaService)
    {
        var curse = FindAllCurse();
        if (curse==null && !curse.Any())
        {
            return [];  
        }

        var lista = new List<CursaDto>();
        foreach (var c in curse)
        {
            var nrPart = participantCursaService.FindNrPartByCursa(c);
            var dto = new CursaDto(c.GetNume(), c.GetCapacitateMinima(), c.GetCapacitateMaxima(), nrPart);
            lista.Add(dto);
        }
        return lista;
    }

}